src/main/java/io/fabric8/elasticsearch/plugin/OpenshiftRequestContextFactory.java

```diff
@@ -90,9 +84,6 @@ public OpenshiftRequestContext create(final RestRequest request) throws Exceptio
         boolean isClusterAdmin = false;
         String user = utils.getUser(request);
         String token = utils.getBearerToken(request);
-        if(utils.hasUserHeader(request) && StringUtils.isBlank(token)) {
```

Comment:
This is the same logic we have in v2.4.4.x.  We no longer need the user header because we no longer effectively need to trust the proxy or an incoming curl request.  We make the assumption if you have a valid token then we will determine who you actually are and allow the request to continue under those findings.  We took it out in the other version because there is no need to validate your real username against the one you say it is if we are always going to use the real one.

Reply:
/lgtm